### PR TITLE
linear cross validation

### PR DESCRIPTION
**Pull request by AshlynPowell**
Added linear option to impute function

### CODE REVIEW

File: de/fitting.py [hunk 1]
```diff
@@ -76,15 +77,22 @@ def impute(data, fitting=False):
 
         # Fit
         if fitting:
-            xx = runOptim(data, niter=500)
-            w, eta = reshapeParams(xx, data.shape[0])
+            if linear:
+                w, prediction = runFitting(data, U)
+            else:    
+                xx = runOptim(data, niter=500)
+                w, eta = reshapeParams(xx, data.shape[0])
```
Comment (Farnazmdi) [inline de/fitting.py:84]: @AshlynPowell I think the `prediction` used here comes from the same dataset used for training. I think you should use the `model.predict()` on the test set. It probably requires some changes to the `Fitting()` function.

**Comment by Farnazmdi** [comment]
@AshlynPowell it appeared the `SSE()` function was not being used so I removed it.
Let me know if you think it is needed.
I also changed a line in test_factorization to assert the corr. coef. when the non-linear model is run is higher than when the linear model is run.
It should be almost ready. You can ask Dr. Meyer for a review.

**Review by Farnazmdi — APPROVED** [review]

**Comment by AshlynPowell** [comment]
@aarmey Could you review this branch for merging? It has the linear model file added, the linear option added to the imputation function and a repeatImputation function that calculates the average correlation coefficient after running the imputation function 20 times.

File: de/factorization.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import numpy as np
 import pandas as pd
+import random
```
Comment (aarmey) [inline de/factorization.py:5]: Don't need this.

File: de/tests/test_factorization.py [hunk 3]
```diff
@@ -81,5 +81,8 @@ def test_split_data():
     data = ImportMelanoma()
     train_X, test_X = split_data(data)
     full_X = impute(train_X)
+    full_X_lin = impute(train_X, linear=True)
 
-    print(ma.corrcoef(ma.masked_invalid(full_X.flatten()), ma.masked_invalid(test_X.flatten())))
+    non_lin = ma.corrcoef(ma.masked_invalid(full_X.flatten()), ma.masked_invalid(test_X.flatten()))[0][1]
```
Comment (aarmey) [inline de/tests/test_factorization.py:86]: Tests should check cases we can be absolutely sure will always be true. I'm not sure about that here.

File: de/linearModel.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from sklearn.linear_model import Lasso
+import numpy as np
+
+def runFitting(data, U=None, max_iter=300000):
```
Comment (aarmey) [inline de/linearModel.py:4]: Probably want an option here to pass in the regularization strength, as we'll want to adjust that in the future.

**Review by aarmey — CHANGES_REQUESTED** [review]

**Review by aarmey — APPROVED** [review]
Looks good.
